apps/wallet-mobile/src/features/Swap/common/SwapFormProvider.tsx

```
@@ -270,9 +335,11 @@ const swapFormReducer = (state: SwapFormState, action: SwapFormAction) => {
         break
 
       case SwapFormActionType.ResetSwapForm:
-      case SwapFormActionType.ClearSwapForm:
         return defaultState
 
+      case SwapFormActionType.ClearSwapForm:
+        return state
```

Isn't this case redundant?

---

It could be removed altogether, yes, but I guess it's there for completeness in case there's something to do at some point?

---

I'd say that everything that updates the form for first should control the change in the state, so if the clear set the inputs and dispatch the reset quantities underlying the state action should be initiated/moved here.

---

owww, it's inside the helpers not inside the state. it should be colocated.

---

k, since we throw instead of defaulting to return the state, I think it's fine, I'd colocate that later.